Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -237,6 +238,71 @@ public AndConstraint<XDocumentAssertions> NotBeEquivalentTo(XDocument unexpected
+        /// child element with the specified <paramref name="expected"/> name.
+        /// </summary>
+        /// <param name="expected">
+        /// The full name <see cref="XName"/> of the expected child element of the current document's Root <see cref="XDocument.Root"/> element.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<XDocumentAssertions> HaveSingleElement(XName expected, string because = "", params object[] becauseArgs)
+        {
+            return HaveElementCount(expected, 1, because, becauseArgs);
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="XDocument.Root"/> element of the current <see cref="XDocument"/> has a the specified <paramref name="count"/> of
+        /// child elements with the specified <paramref name="expected"/> name.
+        /// </summary>
+        /// <param name="expected">
+        /// The full name <see cref="XName"/> of the expected child element of the current document's Root <see cref="XDocument.Root"/> element.
+        /// </param>
+        /// <param name="count">The expected count of elements in the document.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<XDocumentAssertions> HaveElementCount(XName expected, int count, string because = "",
+            params object[] becauseArgs)
+        {
+            if (Subject is null)
+            {
+                throw new InvalidOperationException("Cannot assert the count if the document itself is <null>.");
+            }
+
+            Guard.ThrowIfArgumentIsNull(expected, nameof(expected),
+                "Cannot assert the document has an element count if the element name is <null>*");
+
+            Execute.Assertion
+                .ForCondition(Subject.Root is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith(
+                    "Expected {context:subject} to have root element with child {0}{reason}, but it has no root element.",
+                    expected.ToString());
+
+            var xElements = Subject.Root.Elements(expected);
```

Comment:
If `HaveElementCount` is called inside an `AssertionScope` the assertion above above will not fail immediately, so `Subject.Root` can be null here.
See e.g. `XElementAssertions.HaveValue` for an example of how to handle this.